File: foundation/foundation-transform/foundation-transform-core-java/src/main/java/com/boozallen/data/transform/policy/DataTransformPolicy.java
package com.boozallen.data.transform.policy;

/*-
 * #%L
 * aiSSEMBLE::Foundation::Transform::Java
 * %%
 * Copyright (C) 2021 Booz Allen
 * %%
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 * 
 *      http://www.apache.org/licenses/LICENSE-2.0
 * 
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 * #L%
 */

import org.technologybrewery.mash.BaseMediationManager;

import com.boozallen.aissemble.core.policy.configuration.policy.Policy;

/**
 * {@link DataTransformPolicy} class represents the contract for a data
 * transform policy.
 * 
 * @author Booz Allen Hamilton
 * 
 */
public interface DataTransformPolicy extends Policy {

    /**
     * Returns the mediation manager instance for this policy.
     * 
     * @return mediation manager instance
     */
    public BaseMediationManager getMediationManager();

}

File: foundation/foundation-transform/aissemble-foundation-transform-core-python/src/data_transform_core/policy/data_transform_policy.py
# #%L
# Data Transform::Python::Core
# %%
# Copyright (C) 2021 Booz Allen
# %%
# This software package is licensed under the Booz Allen Public License. All Rights Reserved.
# #L%
###
from policy_manager.policy import Policy
from data_transform_core.mediator import MediationManager


class DataTransformPolicy(Policy):
    """
    DataTransformPolicy class represents the additional policy
    configurations that are needed for data transform.
    """

    mediationManager: MediationManager = MediationManager()
